roxie/ccd/ccdfile.cpp

```diff
@@ -2127,7 +2134,7 @@ class CResolvedFile : implements IResolvedFileCreator, implements ISDSSubscripti
                         {
                             if (mode == RecordTranslationMode::None)
                                 throw MakeStringException(ROXIE_MISMATCH, "Translatable record layout mismatch detected for file %s, but translation disabled", subname);
-                            keyedTranslator.setown(createKeyTranslator(actual->queryRecordAccessor(true), expected->queryRecordAccessor(true)));
+                            keyedTranslator.setown(createKeyTranslator(actual->queryRecordAccessor(true), expected->queryRecordAccessor(true))); // MORE - should this be outside the if? Should I be checking if it is empty?
```

Comment:
I don't know the answer, I think it is ok, but probably should resolve before this is merged.

Reply:
I THINK inside the IF is ok. Hard to see how the keyed part could need translation if the projected does not. Unless we distinguish expected from projected, and actual happens to match projected but not expected I suppose?

As far as checking it is empty - that will be something that needs doing for index but not for disk cases.